backend/wmg/data/config.py

```diff
@@ -0,0 +1,46 @@
+import os
+
+import psutil
+import tiledb
+
+
+def base_config() -> dict:
+    if boto_endpoint_url := os.getenv("BOTO_ENDPOINT_URL"):
+        return {
+            "vfs.s3.endpoint_override": boto_endpoint_url,
+            # localstack does not support S3 virtual addressing (per
+            # https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html)
+            "vfs.s3.use_virtual_addressing": "false",
+        }
+
+    return {
+        "vfs.s3.region": "us-west-2",
+    }
+
+
+def create_ctx(config: dict = {}) -> tiledb.Ctx:
+    cfg = tiledb.Config({**base_config(), **config})
+    ctx = tiledb.Ctx(config=cfg)
+    return ctx
+
+
+def frac_mem(f):
+    mem_size = psutil.virtual_memory().total
+    return int(f * mem_size) // (1024**2) * (1024**2)
```

Comment:
how do you feel about pulling out/naming some of these for clarity/reuse (and the values on line 36 and 40)

Reply:
done! 2aeadb5a